spec/classes/pulp_config_spec.rb

```diff
@@ -72,6 +72,52 @@
     end
   end
 
+  context "with remove_old_repodata configuration" do
+    let :pre_condition do
+      "class {'pulp':
+        enable_rpm              => true,
+        yum_remove_old_repodata_count => 5
+      }"
+    end
+
+    it "should produce valid json" do
+      should contain_file("/etc/pulp/server/plugins.conf.d/yum_distributor.json").with_content(
+        /"remove_old_repodata": True,/
+      ).with_content(
+        /"remove_old_repodata_threshold": 5/
+      ).with({
+        'ensure'    => 'file',
+        'owner'     => 'root',
+        'group'     => 'root',
+        'mode'      => '0644',
+        'show_diff' => false,
+      })
+
+    end
+  end
+
+  context "with nil remove_old_repodata configuration" do
+    let :pre_condition do
+      "class {'pulp':
+        enable_rpm                    => true,
+        yum_remove_old_repodata_count => undef
+      }"
```

Comment:
Setting to undef here doesn't actually work, it defaults to the default.     Is there another strategy to do this?  Some thoughts:
should I have 2 params?  one to enable, and one to set the count, similar to the pulp config

Reply:
I think this will solve it: https://github.com/theforeman/puppet-pulp/pull/415/files#r666904364